hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -1450,14 +1529,23 @@ def draw_float(
         # TODO: consider supporting these float widths at the IR level in the
         # future.
         # width: Literal[16, 32, 64] = 64,
-        # exclude_min and exclude_max handled higher up
+        # exclude_min and exclude_max handled higher up,
+        forced: Optional[float] = None,
     ) -> float:
         assert smallest_nonzero_magnitude > 0
+        assert not math.isnan(min_value)
+        assert not math.isnan(max_value)
+
+        if forced is not None:
+            assert not math.isnan(forced)
+            assert min_value <= forced <= max_value
```

Comment:
Reading this back, I'm realizing that I don't think there's any reason we can't support `forced=nan`. I'll look at this tomorrow.

Reply:
I've added this.